MVVM-C/Controllers/Article Details/DetailViewVC.swift

```diff
@@ -0,0 +1,166 @@
+        self.imageArticle.downloadThumbnail(link)
+    }
+    
+    private func configureNavBar() {
+        let barButtonSystemItem: UIBarButtonItem.SystemItem = (self.viewModel?.isFavorited == true ? .trash : .add)
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: barButtonSystemItem, target: self, action: #selector(addToFavorites))
+    }
+    
+    @objc func addToFavorites() {
+        self.viewModel?.setFavorite()
+    }
+    
+    private func configureUI() {
+        view.addSubview(articleLabel)
+        view.addSubview(abstractLabel)
+        view.addSubview(imageArticle)
+        view.addSubview(safariButton)
+        
+        
+        NSLayoutConstraint.activate([
+            articleLabel.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 20),
+            articleLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            articleLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            articleLabel.heightAnchor.constraint(equalToConstant: 40),
+            
+            abstractLabel.topAnchor.constraint(equalTo: articleLabel.bottomAnchor, constant: 10),
+            abstractLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            abstractLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            abstractLabel.heightAnchor.constraint(equalToConstant: 100),
+            
+            imageArticle.topAnchor.constraint(equalTo: abstractLabel.bottomAnchor, constant: 10),
+            imageArticle.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            imageArticle.widthAnchor.constraint(equalToConstant: 300),
+            imageArticle.heightAnchor.constraint(equalToConstant: 300),
+            
+            safariButton.topAnchor.constraint(equalTo: imageArticle.bottomAnchor, constant: 40),
+            safariButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            safariButton.heightAnchor.constraint(equalToConstant: 50),
+            safariButton.widthAnchor.constraint(equalToConstant: 250)
+        ])
+    }
+}
+
+
+class GoTLabel: UILabel {
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configure()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    convenience init(textAlignment: NSTextAlignment, fontSize: CGFloat) {
+        self.init(frame: .zero)
+        self.textAlignment = textAlignment
+        self.font = UIFont.systemFont(ofSize: fontSize, weight: .bold)
+    }
+    
+    func configure() {
+        textColor = .label
+        translatesAutoresizingMaskIntoConstraints = false
+        lineBreakMode = .byTruncatingTail
+        adjustsFontForContentSizeCategory = true
+        numberOfLines = 4
+    }
+}
+
+
+class GoTImageThumbnail: UIImageView {
```

Comment:
why this GoTImageThimbnail class is in DetailsViewVC?

Reply:
Also just copied that class to go fast